### PR TITLE
Stabilize macro_vis_matcher

### PR DESCRIPTION
**Pull request by jkozlowski**
This PR should stabilize [macro_vis_matcher](https://github.com/rust-lang/rust/issues/41022) feature. 

- [x] "reference" book changes: https://github.com/rust-lang-nursery/reference/pull/400
- [x] "Rust by example" book changes: https://github.com/rust-lang/rust-by-example/pull/1096
- [x] "clippy" changes: https://github.com/rust-lang-nursery/rust-clippy/pull/3055

r? @cramertj

### CODE REVIEW

File: src/librustc_data_structures/lib.rs [hunk 1]
```diff
@@ -25,7 +25,6 @@
 #![feature(unsize)]
 #![feature(specialization)]
 #![feature(optin_builtin_traits)]
-#![feature(macro_vis_matcher)]
```
[the remark this answers is not in the document]
Comment (jkozlowski) [reply src/librustc_data_structures/lib.rs]: Duh, reading comprehension not my best suit apparently. Hopefully fixed up now, let's see what CI says

**Comment by jkozlowski** [comment]
@cramertj should be better now :)

File: src/libsyntax/feature_gate.rs [hunk 2]
```diff
@@ -648,6 +645,8 @@ declare_features! (
     (accepted, repr_transparent, "1.28.0", Some(43036), None),
     // Defining procedural macros in `proc-macro` crates
     (accepted, proc_macro, "1.29.0", Some(38356), None),
+    // Allows use of the :vis macro fragment specifier
+    (accepted, macro_vis_matcher, "1.29.0", Some(41022), None),
```
Comment (mati865) [inline src/libsyntax/feature_gate.rs]: It should be `1.30.0`

Comment (jkozlowski) [reply src/libsyntax/feature_gate.rs]: Oops sorry: I wondered about this after rebasing on top of master, but forge instructions said to look at forge for the next stable and that was 1.29.

Shall I send a PR to fix this up?

Comment (mati865) [reply src/libsyntax/feature_gate.rs]: Pinging reviewer: @cramertj

Comment (cramertj) [reply src/libsyntax/feature_gate.rs]: Sure-- a PR to bump the version number would be appreciated :smile:

Comment (cramertj) [reply src/libsyntax/feature_gate.rs]: Sorry for missing this the first time around.

Comment (jkozlowski) [reply src/libsyntax/feature_gate.rs]: @cramertj sorry about that, here's the PR: https://github.com/rust-lang/rust/pull/53702

**Pull request merged**